src/app/modules/main/module.nim

```diff
@@ -508,6 +511,27 @@ method getCommunitySectionModule*[T](self: Module[T], communityId: string): QVar
 
   return self.communitySectionsModule[communityId].getModuleAsVariant()
 
+method rebuildChatSearchModel*[T](self: Module[T]) =
+  let transformItem = proc(item: chat_section_base_item.BaseItem, sectionId, sectionName: string): chat_search_item.Item =
+    result = chat_search_item.initItem(item.id(), item.name(), item.color(), item.icon(), sectionId, sectionName)
```

Comment:
You can make `transformItem` an actual proc if you want. Putting a proc inside another proc like that is not something we do often

Reply:
In this case, I would like to keep the procedure inside `rebuildChatSearchModel`, because If I move it outside it will lose its context. It's also a few-liner so the readability still stays on the high level.